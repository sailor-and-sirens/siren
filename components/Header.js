import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { connect } from 'react-redux';
import { actionCreators } from '../actions';
import { headerActions } from '../actions/Header';
import { MaterialIcons } from '@expo/vector-icons';
import ModalComponent from './Modal';
import Filter from './Filter';

const mapStateToProps = (state) => ({
  view: state.header.view
})

class Header extends Component {

  render() {
    let filter = null;
    if (this.props.view === 'Inbox' ||
        this.props.view === 'Playlists') {
      filter = (<View>
        <ModalComponent>
          <Filter/>
        </ModalComponent>
        <MaterialIcons style={styles.navigationLink} onPress={() => this.props.dispatch(actionCreators.toggleModal())} size={30} name='filter-list' />
      </View>)
    }
    return (
      <View>
        <Text style={styles.header}>{this.props.view}</Text>
        <View style={styles.navigation}>
          <MaterialIcons style={styles.navigationLink} onPress={() => {
            this.props.dispatch(actionCreators.updatePlaylistFilter('All'));
            this.props.dispatch(actionCreators.updateFilters());
            this.props.dispatch(headerActions.changeView('Inbox'));
          }} size={30} name='inbox'/>
          <MaterialIcons style={styles.navigationLink} onPress={() => this.props.dispatch(headerActions.changeView('Playlists'))} size={30} name='playlist-play'/>
          <MaterialIcons style={styles.navigationLink} onPress={() => this.props.dispatch(headerActions.changeView('Search'))} size={30} name='search'/>
          {filter}
          <MaterialIcons style={styles.navigationLink} onPress={() => this.props.dispatch(headerActions.changeView('Settings'))} size={30} name='settings' />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  header: {
    marginTop: 0,
    padding: 10,
    paddingTop: 30,
    fontSize: 16,
    fontWeight: 'bold',
    textAlign: 'center',
    color: '#000',
    backgroundColor: '#50BFB9'
  },
  navigation: {
    alignSelf: 'stretch',
    flexDirection: 'row',
    justifyContent: 'space-around',
    backgroundColor: '#BCDDD4',
  },
  navigationLink: {
    padding: 20,
    paddingTop: 8,
    paddingBottom: 8,
    color: '#000',
    textShadowColor: '#999',
    textShadowOffset: {width: 1.25, height: 1.25},
    textShadowRadius: 2
  }
});

export default connect(mapStateToProps)(Header);
