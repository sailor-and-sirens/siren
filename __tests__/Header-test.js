import React from 'react';
import Header from '../components/Header.js';
import header from '../reducers/Header.js';
import { headerActions } from '../actions/Header.js';
import { initialState } from '../reducers/Header.js';
import * as renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from '../reducers/index';
const store = createStore(reducer);

test('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Header />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

//TESTING REDUCERS
test('reducer returns default state and renders correctly', () => {
  expect(header(initialState, {type: '_NULL'})).toMatchSnapshot();
  //Alternative to snapshots
    // expect(header(initialState, {type: '_NULL'})).toEqual({
    //   ...initialState,
    //   view: 'Inbox'
    // });
});
test('reducer changes state and renders correctly', () => {
  expect(header(initialState, {type: 'CHANGE_VIEW', payload: 'Filter'})).toMatchSnapshot();
});

//TESTING ACTIONS
test('Creates changeview action for Inbox', () => {
  expect(headerActions.changeView('Inbox')).toMatchSnapshot();
});
test('Creates changeview action for Playlist', () => {
  expect(headerActions.changeView('Playlist')).toMatchSnapshot();
});
