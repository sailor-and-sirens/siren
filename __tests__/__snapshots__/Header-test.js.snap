// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates changeview action for Inbox 1`] = `
Object {
  "payload": "Inbox",
  "type": "CHANGE_VIEW",
}
`;

exports[`Creates changeview action for Playlist 1`] = `
Object {
  "payload": "Playlist",
  "type": "CHANGE_VIEW",
}
`;

exports[`reducer changes state and renders correctly 1`] = `
Object {
  "view": "Filter",
}
`;

exports[`reducer returns default state and renders correctly 1`] = `
Object {
  "view": "Authentication",
}
`;

exports[`renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#50BFB9",
        "color": "#000",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginTop": 0,
        "padding": 10,
        "paddingTop": 30,
        "textAlign": "center",
      }
    }
  >
    Authentication
  </Text>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#BCDDD4",
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</View>
`;
